encoding/json/encoding_test.go

```diff
@@ -1114,25 +1140,35 @@ func convertValueFromScript(t *testing.T, script string) cadence.Value {
 	return value
 }
 
-func testAllEncode(t *testing.T, tests ...encodeTest) {
+func testAllEncodeAndDecode(t *testing.T, tests ...encodeTest) {
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			testEncode(t, test.val, test.expected)
+			testEncodeAndDecode(t, test.val, test.expected)
 		})
 	}
 }
 
-func testEncode(t *testing.T, val cadence.Value, expectedJSON string) {
```

Comment:
Split this function into separate encode and decode steps, renamed to `testEncodeAndDecode`. Links and references are only tested with `testEncode`.